src/model/neighborlist_modes.h

```diff
@@ -0,0 +1,208 @@
+					idxZ);
+
+			real3_t fAcc = {0.0,0.0,0.0};
+			REAL eAcc = 0.0;
+
+			real3_t fRec = {0.0,0.0,0.0};
+			REAL eRec = 0.0;
+
+			for (unsigned j = 0; j < nDesc.numEl; ++j) {
+				const unsigned nIdx = grid->getNeighbor(nDesc.idx + j);
+
+				REAL dx = posLigX - RecPosX[nIdx];
+				REAL dy = posLigY - RecPosY[nIdx];
+				REAL dz = posLigZ - RecPosZ[nIdx];
+
+				fRec.x = 0.0;
+				fRec.y = 0.0;
+				fRec.z = 0.0;
+				eRec  = 0.0;
+
+
+				const REAL dr2 = dx * dx + dy * dy + dz * dz;
+
+				if (grid->outOfPlateau(dr2)) {
+					continue;
+				}
+
+				const REAL dr2_inv = 1.0/dr2; // inverse of dr2
+
+				// Scale distances
+				dx *= dr2_inv;
+				dy *= dr2_inv;
+				dz *= dr2_inv;
+
+				const size_t atomTypeRec = rec->type()[nIdx];
+
+				// calculate energy and potential/energy of LJ/VdW potential
+				assert(atomTypeRec > 0);
+				assert(atomTypeRec < 99);
+				assert(atomTypeLig > 0);
+				assert(atomTypeLig < 99);
+
+				real3_t fVdW;
+				REAL eVdW;
+
+				REAL swi = 1;
+				REAL swiPlateau = 1;
+				LJPotForce(dr2, dr2_inv, dx, dy, dz,
+						table->getParams(atomTypeRec-1, atomTypeLig-1),
+						swi, table->potShape(),
+						fVdW.x, fVdW.y, fVdW.z, eVdW);
+
+				fAcc.x  += fVdW.x;
+				fAcc.y  += fVdW.y;
+				fAcc.z  += fVdW.z;
+				eAcc += eVdW;
+
+				fRec.x -= fVdW.x;
+				fRec.y -= fVdW.y;
+				fRec.z -= fVdW.z;
+				eRec  +=  eVdW;
+
+				const REAL chargeLig = lig->charge()[i];
+				const REAL chargeRec = rec->charge()[nIdx];
+
+				const REAL chargeLigRec = chargeLig * chargeRec * simParam->ffelec;
+				const bool calc_elec = abs(chargeLigRec) > 0.001; // evaluate electric potential
+
+				REAL rdx, rdy, rdz;
+				REAL ratio = grid->getRatio(dr2);
+				rdx = ratio*dx;
+				rdy = ratio*dy;
+				rdz = ratio*dz;
+
+				LJPotForce(grid->dPlateau2(), grid->dPlateau2_inv(), rdx, rdy, rdz,
+					table->getParams(atomTypeRec-1, atomTypeLig-1),
+					swiPlateau, table->potShape(),
+					fVdW.x, fVdW.y, fVdW.z, eVdW);
+
+				fAcc.x  -= fVdW.x;
+				fAcc.y  -= fVdW.y;
+				fAcc.z  -= fVdW.z;
+				eAcc -= eVdW;
+
+				fRec.x += fVdW.x;
+				fRec.y += fVdW.y;
+				fRec.z += fVdW.z;
+				eRec  -=  eVdW;
+
+
+				if (calc_elec) {
+					REAL eEl;
+					real4_t fEl;
+
+					// calculate energy and potential/energy of charge potential
+					ChargePotForce(dr2_inv, dx, dy, dz,
+							chargeLigRec,
+							swi, simParam->dielec,
+							fEl.x, fEl.y, fEl.z, eEl);
+
+					fAcc.x += fEl.x;
+					fAcc.y += fEl.y;
+					fAcc.z += fEl.z;
+					eAcc += eEl;
+
+					fRec.x -= fEl.x;;
+					fRec.y -= fEl.y;
+					fRec.z -= fEl.z;
+					eRec  +=  eEl;
+
+					ChargePotForce(grid->dPlateau2_inv(), rdx, rdy, rdz,
+							chargeLigRec,
+							swiPlateau, simParam->dielec,
+							fEl.x, fEl.y, fEl.z, eEl);
+					fAcc.x -= fEl.x;
+					fAcc.y -= fEl.y;
+					fAcc.z -= fEl.z;
+					eAcc -= eEl;
+
+					fRec.x += fEl.x;;
+					fRec.y += fEl.y;
+					fRec.z += fEl.z;
+					eRec  -=  eEl;
+
+
+				}
+
+				outRec_fx[nIdx]+=fRec.x;
```

Comment:
code style: please put spaces before and after the operator: outRec_fx[nIdx] += fRec.x;